### PR TITLE
Fix | FailoverPartner key check in SqlConnectionPoolGroupProviderInfo

### PR DESCRIPTION
**Pull request by lcheunglci**
Relates to issue #1594 and #1613 , I think we can either do a check if the key exists first before retrieving the value from the SqlConnectionString using the Dictionary get [operator[]](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.item?view=net-6.0) or use [TryGetValue ](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-6.0) with null as the default value if that's the expected behaviour because I don't think it's obvious that we need to specify `FailOverPartner` set it to an empty string.

### CODE REVIEW

**Comment by ErikEJ** [comment]
Wonder if a test is possible - why does this crop up?

**Comment by lcheunglci** [comment]
I think it depends on how the Sql Server is configured, so a special test agent/vm might be required in the build/test pipeline to be able to verify this properly.

**Review by David-Engel — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolGroupProviderInfo.cs [hunk 1]
```diff
@@ -104,7 +104,8 @@ private PermissionSet CreateFailoverPermission(SqlConnectionString userConnectio
             //       the server, we will use that name over what was specified  
             //       in the original connection string.
 
-            if (null == userConnectionOptions[SqlConnectionString.KEY.FailoverPartner])
+            if (userConnectionOptions.ContainsKey(SqlConnectionString.KEY.FailoverPartner) &&
```
[the remark this answers is not in the document]
Comment (mark-raymond) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolGroupProviderInfo.cs:107]: At least in #1594, the data source was an AG. It's not specified in #1613 whether the data source was an AG.

[the remark this answers is not in the document]
Comment (gregpakes) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolGroupProviderInfo.cs:107]: I have just experienced this issue in a non-AG setup.

**Comment by gregpakes** [comment]
Is this getting patched into v4?

**Pull request merged**